web/src/message_edit_history.ts

```diff
@@ -18,30 +21,58 @@ import * as timerender from "./timerender";
 import * as ui_report from "./ui_report";
 import {user_settings} from "./user_settings";
 
-export function fetch_and_render_message_history(message) {
-    channel.get({
+type ContentEditHistoryItem = {
+    timestamp: string;
+    display_date: string;
+    show_date_row: boolean;
+    edited_by_notice: string;
+    body_to_render?: string;
+    topic_edited?: boolean;
+    prev_topic?: string;
+    new_topic?: string;
+    stream_changed?: boolean;
+    prev_stream?: string;
+};
+
+const fetch_and_render_message_history_response_schema = z.object({
+    message_history: z.array(
+        z.object({
+            timestamp: z.number(),
+            user_id: z.number(),
+            rendered_content: z.string(),
+            prev_topic: z.string(),
+            prev_content: z.string(),
+            content_html_diff: z.string(),
+            topic: z.string(),
+            prev_stream: z.number(),
+        }),
+    ),
+});
```

Comment:
I'm having a hard time lining this type definition up with https://zulip.com/api/get-message-history. Maybe part of the problem is the order of the fields you have feels basically random, neither alphabetical nor matching the API docs?

Reply:
I forgot to use the API docs, sorry about that. I'll try to remember that as part of using `zod` for future typescript conversions. Thanks!